engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/br/CNPJValidator.java

```diff
@@ -55,6 +56,13 @@ public boolean isValid(CharSequence value, ConstraintValidatorContext context) {
 			return true;
 		}
 
+		// Check for repeated digits
+		if ( value.toString().chars()
```

Comment:
Maybe let's replace this with a simple for loop? Something along the lines:
```java
// the number is too short, hence it's invalid anyway:
if ( value.length() < 2 ) {
	return false;
}

char firstDigit = value.charAt( 0 );
char otherDigit = value.charAt( 1 );
for ( int i = 2; i < value.length(); i++ ) {
	char c = value.charAt( i );
	if ( Character.isDigit( c ) && firstDigit != c ) {
		otherDigit = c;
	}
}
if ( firstDigit == otherDigit ) {
	return false;
}
```

From what I see `distinct` may not be optimized:
```java
public final IntStream distinct() {
	// While functional and quick to implement, this approach is not very efficient.
	// An efficient version requires an int-specific map/set implementation.
	return boxed().distinct().mapToInt(i -> i);
}
```

Reply:
perfect, I'll change it. Thanks!